src/Common/tests/System/RealFormatterTestsBase.netcoreapp.cs

```diff
@@ -14,29 +14,118 @@ public abstract class RealFormatterTestsBase
         protected abstract string InvariantToStringDouble(double d, string format);
         protected abstract string InvariantToStringSingle(float f, string format);
 
+        [Theory]
+        [InlineData("#,###.00", double.Epsilon, ".00")]
+        [InlineData("#,###.00", double.MaxValue, "179,769,313,486,232,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000.00")]
```

Comment:
As called out in https://github.com/dotnet/coreclr/pull/25096 and related issues, custom-numeric format strings are mostly backwards compatible in 3.0.

I plan on having a fix for 5.0 so that there is longer lead time on the testing there (mostly it just involves processing the format string a bit up front to determine if it involves significant digits or decimal digits, and how many digits are being requested).